src/test/java/com/example/project/test/EmployeeTest.java

```diff
@@ -0,0 +1,127 @@
+                        .profileImage(StringUtils.EMPTY)
+                        .build()},
+                {EmployeeData.builder()
+                        .id(3)
+                        .employeeName("Ashton Cox")
+                        .employeeSalary(86000)
+                        .employeeAge(66)
+                        .profileImage(StringUtils.EMPTY)
+                        .build()},
+        };
+    }
+
+    @Test(dataProvider = "existingEmployees")
+    public void checkGetEmployeesByIds(EmployeeData employeeData) {
+        val employee = getEmployee(employeeData.getId());
+
+        assertThat(employee.getData()).isEqualTo(employeeData);
+    }
+
+    @Test
+    public void checkForEmployeeWithNotExistingId() {
+        val notExistingId = 25;
+        val employee = getEmployee(notExistingId);
+
+        val softly = new SoftAssertions();
+        softly.assertThat(employee.getData()).isNull();
+        softly.assertThat(employee.getStatus()).isEqualTo("failure");
+        softly.assertThat(employee.getMessage()).isEqualTo("Failure! There is no such employee.");
+        softly.assertAll();
+    }
+
+    @Test
+    public void testCreateNewEmployee() {
+        val newEmployeeData = NewEmployeeData.builder()
+                .name("Adam")
+                .salary(5000)
+                .age(40)
+                .build();
+
+        val employeeToBeCreated = postEmployee(newEmployeeData);
+
+        val newEmployeeId = employeeToBeCreated.getData().getId();
+
+        val createdEmployee = getEmployee(newEmployeeId);
+
+        assertThat(createdEmployee.getData()).as("check if new employee has not null data").isNotNull();
+
+        val softly = new SoftAssertions();
+        softly.assertThat(employeeToBeCreated.getMessage()).isEqualTo("Successfully! Record has been added.");
+        softly.assertThat(createdEmployee.getData().getId()).isEqualTo(employeeToBeCreated.getData().getId());
+        softly.assertThat(createdEmployee.getData().getEmployeeName()).isEqualTo(employeeToBeCreated.getData().getName());
+        softly.assertThat(createdEmployee.getData().getEmployeeAge()).isEqualTo(employeeToBeCreated.getData().getAge());
+        softly.assertThat(createdEmployee.getData().getEmployeeSalary()).isEqualTo(employeeToBeCreated.getData().getSalary());
+        softly.assertAll();
+    }
+
+    private Employee getEmployee(int id) {
+        return given()
+                    .spec(requestSpecification)
+                    .pathParams("id", id)
+                .expect()
+                    .spec(responseSpecification)
+                .when()
+                    .get("/employee/{id}")
+                .as(Employee.class);
+    }
+
+    private NewEmployee postEmployee(NewEmployeeData newEmployeeData) {
+        return given()
+                    .spec(requestSpecification)
+                    .contentType("application/json")
+                    .body(new Gson().toJson(newEmployeeData))
```

Comment:
I would suggest to create a final gson instance at the class level during initialisation and reuse it all the time instead of creating a new one for each call